src/components/reusable/header/Menu.tsx

```diff
@@ -0,0 +1,185 @@
+import React, { useEffect, useState } from 'react';
+import { Button } from '@mui/material';
+import ReceiptLongOutlinedIcon from '@mui/icons-material/ReceiptLongOutlined';
+import PersonOutlinedIcon from '@mui/icons-material/PersonOutlined';
+import SettingsOutlinedIcon from '@mui/icons-material/SettingsOutlined';
+import InfoOutlinedIcon from '@mui/icons-material/InfoOutlined';
+import FavoriteBorderOutlinedIcon from '@mui/icons-material/FavoriteBorderOutlined';
+import LogoutOutlinedIcon from '@mui/icons-material/LogoutOutlined';
+import SupportAgentOutlinedIcon from '@mui/icons-material/SupportAgentOutlined';
+import ChevronRightOutlinedIcon from '@mui/icons-material/ChevronRightOutlined';
+import { useRouter } from 'next/router';
+import { ROUTES } from 'src/routes';
+import { removeToken } from 'src/redux/slices/tokenSlice';
+import { removeUser } from 'src/redux/slices/userSlice';
+import { useLocales } from 'src/locales';
+import { useAppDispatch, useTypedSelector } from 'src/redux/store';
+import { TRANSACTION_EXAMPLE, USER_ROLE } from 'src/constants';
+import { useTopUpMutation, useWithdrawMutation } from 'src/redux/api/paymentApi';
+import { useGetUserInfoQuery } from 'src/redux/api/userApi';
+import {
+  Arrow,
+  BalanceAmount,
+  BalanceBlock,
+  BalanceParagraph,
+  BalanceTitle,
+  MenuListItem,
+  MenuItemStyled,
+  OperationButton,
+  StyledMenu,
+} from './styles';
+
+interface MenuDropdownProps {
+  children: React.ReactNode;
+  onClick: () => void;
+}
+
+const MenuDropdown: React.FC<MenuDropdownProps> = ({ children, onClick }): JSX.Element | null => {
+  const router = useRouter();
+  const { translate } = useLocales();
+  const dispatch = useAppDispatch();
+  const [withdraw] = useWithdrawMutation();
+  const [topUp] = useTopUpMutation();
+
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const [balance, setBalance] = useState<number>(0);
+  const open = Boolean(anchorEl);
+  const user = useTypedSelector((state) => state.user.user);
+  const { data: userInfo } = useGetUserInfoQuery(user?.id);
+
+  useEffect(() => {
+    if (userInfo?.balance !== undefined) {
+      setBalance(userInfo.balance);
+    }
+  }, [userInfo]);
+
+  if (!user) return null;
+
+  const { role } = user;
+
+  const handleClick = (event: React.MouseEvent<HTMLElement>): void => {
+    setAnchorEl(event.currentTarget);
+    onClick();
+  };
+
+  const handleClose = (): void => {
+    setAnchorEl(null);
+  };
+
+  const handleLogOut = async (): Promise<void> => {
+    try {
+      dispatch(removeToken());
+      dispatch(removeUser());
+      handleClose();
+      window.location.reload();
```

Comment:
I read prev conversation about reloading. You can just redirect user to login page without reloading.